pkg/operation/cloudbotanist/packetbotanist/addons.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package packetbotanist
+
+import (
+	"github.com/gardener/gardener/pkg/operation/common"
+)
+
+// DeployKube2IAMResources - Not needed on Packet
+func (b *PacketBotanist) DeployKube2IAMResources() error {
+        return nil
+}
+
+// DestroyKube2IAMResources - Not needed on Packet.
+func (b *PacketBotanist) DestroyKube2IAMResources() error {
+        return nil
+}
+
+// GenerateKube2IAMConfig - Not needed on Packet.
+func (b *PacketBotanist) GenerateKube2IAMConfig() (map[string]interface{}, error) {
+        return common.GenerateAddonConfig(nil, false), nil
+}
+
+// GenerateStorageClassesConfig generates values which are required to render the chart storage-classes properly.
+func (b *PacketBotanist) GenerateStorageClassesConfig() (map[string]interface{}, error) {
+        return map[string]interface{}{
+                "StorageClasses": []map[string]interface{}{
+                        {
+                                "Name":           "csi-packet-standard",
+                                "IsDefaultClass": true,
+                                "Provisioner":    "net.packet.csi",
+                                "Parameters": map[string]interface{}{
+                                        "type": "standard",
+                                },
+                        },
+                        {
+                                "Name":           "csi-packet-performance",
+                                "IsDefaultClass": false,
+                                "Provisioner":    "net.packet.csi",
+                                "Parameters": map[string]interface{}{
+                                        "plan": "performance",
+                                },
+                        },
+                },
+        }, nil
+}
+
+// GenerateNginxIngressConfig generates values which are required to render the chart nginx-ingress properly.
+func (b *PacketBotanist) GenerateNginxIngressConfig() (map[string]interface{}, error) {
+        return common.GenerateAddonConfig(nil, b.Shoot.NginxIngressEnabled()), nil
+}
+
+// to ease working with the returned values from GenerateVPNShootConfig
```

Comment:
hm, I don't get it, why not using `map[string]interface{}` directly instead of creating a new type?

Reply:
Honestly? It just made it cleaner and easier to work with, and had no negative impact, since the compilation nets the same result anyways. If found it easier to read:

```go
func (b *PacketBotanist) GenerateVPNShootConfig() (map[string]interface{}, error) {
        kv := keyvalue{
                "image": "packethost/metabot:latest", "command": "ip 4 private parent network",
        }
        config := keyvalue{
                "initContainers": []keyvalue{kv},
        }
        return config, nil
}
```

Than:

```go
func (b *PacketBotanist) GenerateVPNShootConfig() (map[string]interface{}, error) {
        kv := map[string]interface{}{
                "image": "packethost/metabot:latest", "command": "ip 4 private parent network",
        }
        config := map[string]interface{}{
                "initContainers": []map[string]interface{}{kv},
        }
        return config, nil
}
```

But I don't care much in the end. Removing it.